### ISSUE TITLE
ANR from MediaDrm API access with 4.10.3

### ISSUE THREAD

**Issue by chao2zhang** [segmentio/analytics-android#789]
Hello folks, recently we upgraded to 4.10.3 but the MediaDrm API is being slow for some devices.

```
io.sentry.android.core.ApplicationNotResponding: Application Not Responding for at least 5000 ms.
    at android.media.MediaDrm.native_setup(MediaDrm.java)
    at android.media.MediaDrm.<init>(MediaDrm.java:293)
    at com.segment.analytics.internal.Utils.getUniqueID
    at com.segment.analytics.internal.Utils.getDeviceId
    at com.segment.analytics.AnalyticsContext.putDevice(AnalyticsContext.java:239)
    at com.segment.analytics.AnalyticsContext.create
    at com.segment.analytics.Analytics$Builder.build
    at com.instacart.client.api.analytics.ICAnalyticsServiceImpl$segmentAnalytics$2.invoke(ICAnalyticsServiceImpl.kt:108)
    at com.instacart.client.api.analytics.ICAnalyticsServiceImpl$segmentAnalytics$2.invoke(ICAnalyticsServiceImpl.kt:102)
    at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
   ...
```

I am wondering if 
- 1. it is possible to leverage other APIs to get identifier 
- 2. It is possible to turn off or supply our own unique ID.

**Comment by wenxi-zeng** [comment]
hey @chao2zhang thanks for reporting this issue. unfortunately, MediaDrm is the only solution we can find to replace a unresettable id and it is also recommended alternative by Google. 

but yes, you can extend `AnalyticsContext` and override the `putDevice` method to supply with y our own unique id. 

feel free to reopen this issue if you found a customized `AnalyticsContext` does not work.

**Comment by wenxi-zeng** [comment]
hey @chao2zhang just fyi, we released a new version 4.10.4 that queries drm api asynchorizely, which should resolve the ANR issue
